### PR TITLE
 [IMP]HR :Set a user as follower when Manager encode something in holida...

### PR DESCRIPTION
**Pull request by ara-odoo**
...y and expense .
https://www.odoo.com/web#id=10455&view_type=form&model=project.task&action=327
pad: https://pad.odoo.com/p/r.SeseD2mZFpiLBNT8

### CODE REVIEW

File: addons/hr_expense/hr_expense.py [hunk 1]
```diff
@@ -112,6 +112,25 @@ def _get_currency(self, cr, uid, context=None):
...
+                self.message_subscribe_users(cr, uid, [hr_expense_id], user_ids=[employee.user_id.id], context=context)
+        return hr_expense_id
+
+    def write(self, cr, uid, ids, vals, context=None):
+        employee_id = vals.get('employee_id')
+        if employee_id:
+            employee = self.pool.get('hr.employee').browse(cr, uid, employee_id, context=context)
+            if employee.user_id:
+                self.message_subscribe_users(cr, uid, ids, user_ids=[employee.user_id.id], context=context)
+        return super(hr_expense_expense, self).write(cr, uid, ids, vals, context=context)
```
Comment (vme-odoo) [inline addons/hr_expense/hr_expense.py]: make a method and move duplicate code.

Comment (vme-odoo) [inline addons/hr_expense/hr_expense.py]: first complete write method afterwards do other stuff.

File: addons/hr_holidays/hr_holidays.py [hunk 2]
```diff
@@ -357,12 +357,23 @@ def create(self, cr, uid, values, context=None):
...
-        return super(hr_holidays, self).create(cr, uid, values, context=context)
+        return hr_holiday_id
 
     def write(self, cr, uid, ids, vals, context=None):
+        employee_id = vals.get('employee_id')
+        if employee_id:
+            employee = self.pool.get('hr.employee').browse(cr, uid, employee_id, context=context)
+            if employee.user_id:
+                self.message_subscribe_users(cr, uid, ids, user_ids=[employee.user_id.id], context=context)
         if vals.get('state') and vals['state'] not in ['draft', 'confirm', 'cancel'] and not self.pool['res.users'].has_group(cr, uid, 'base.group_hr_user'):
```
Comment (vme-odoo) [inline addons/hr_holidays/hr_holidays.py]: same as above comments.

File: addons/hr_expense/hr_expense.py [hunk 3]
```diff
@@ -112,6 +112,26 @@ def _get_currency(self, cr, uid, context=None):
...
     }
 
+    def add_follower(self, cr, uid, ids,employee_id, context=None):
+        employee = self.pool.get('hr.employee').browse(cr, uid, employee_id, context=context)
+        if  employee and employee.user_id:
+            self.message_subscribe_users(cr, uid, ids, user_ids=[employee.user_id.id], context=context)
+        return True
+
+    def create(self, cr, uid, values, context=None):
+        if context is None:
```
Comment (rim-odoo) [inline addons/hr_expense/hr_expense.py]: use `if not context`, else you will create a new `{}` in the case where context is already `{}`

File: addons/hr_expense/hr_expense.py [hunk 4]
```diff
@@ -112,6 +112,26 @@ def _get_currency(self, cr, uid, context=None):
         'currency_id': _get_currency,
     }
 
+    def add_follower(self, cr, uid, ids,employee_id, context=None):
```
Comment (rim-odoo) [inline addons/hr_expense/hr_expense.py]: - If this method is meant to be used internally, prefix its name with an `_`.
-

**Comment by rim-odoo** [comment]
LGTM

**Pull request merged**